src/plugins/google/ssh.ts

```diff
@@ -35,4 +41,36 @@ export const gcpSshProvider: SshProvider<
       ),
     },
   }),
+  cloudProviderLogin: async () => undefined, // TODO @ENG-2284 support login with Google Cloud
+  proxyCommand: (request) => {
+    return [
+      "gcloud",
+      "compute",
+      "start-iap-tunnel",
+      request.id,
+      "%p",
+      // --listen-on-stdin flag is required for interactive SSH session.
+      // It is undocumented on page https://cloud.google.com/sdk/gcloud/reference/compute/start-iap-tunnel
+      // but mention on page https://cloud.google.com/iap/docs/tcp-by-host
+      // and also found in `gcloud ssh --dry-run` output
+      "--listen-on-stdin",
+      `--zone=${request.zone}`,
+      `--project=${request.projectId}`,
+    ];
+  },
+  reproCommands: () => undefined, // TODO @ENG-2284 support login with Google Cloud
+  preTestAccessPropagationArgs: (cmdArgs) => {
+    if (cmdArgs.sudo || ("command" in cmdArgs && cmdArgs.command === "sudo")) {
```

Comment:
nit: is this sudo check happening anywhere else if so, we could use a helper function like: `isSudoCommand(cmdArgs)`